Lib/test/test_import/__init__.py

```diff
@@ -2701,6 +2702,43 @@ def test_pyimport_addmodule_create(self):
         mod = _testcapi.check_pyimport_addmodule(name)
         self.assertIs(mod, sys.modules[name])
 
+class TestImportAccented(unittest.TestCase):
+    # XXX: There should be tests with PYTHONCASEOK as well
+    #      (for those platforms where this is relevant)
+    dir_name = os.path.abspath(TESTFN)
+
+    def setUp(self):
+        self.sys_path = sys.path[:]
+        os.mkdir(self.dir_name)
+        sys.path.insert(0, self.dir_name)
+        importlib.invalidate_caches()
+
+    def tearDown(self):
+        sys.path[:] = self.sys_path
+        importlib.invalidate_caches()
+        rmtree(self.dir_name)
+
+    def assert_importing_possible(self, name):
+        filename = os.path.join(self.dir_name, f"{name}.py")
+        with open(filename, "w") as stream:
+            stream.write("SPAM = 'spam'\n")
+
+        values = {}
+        exec(f"from {name} import SPAM", values, values)
+        self.assertEqual(values["SPAM"], "spam")
+        del sys.modules[unicodedata.normalize('NFKC', name)]
```

Comment:
Also, I'm not worried about `KeyError` here, since the point of this change is that we expect it to be set.  However, it might make sense to be explicit about that expectation:

```suggestion
        self.assertIn(normalized, sys.modules)
        del sys.modules[normalized]
```

(`normalized = unicodedata.normalize('NFKC', name)` is at the top of the method.)